--- eclipse/Clean/src/main/java/net/minecraft/server/ServerEula.java	2019-01-11 20:13:52.865721800 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/server/ServerEula.java	2019-01-11 20:13:26.787466800 -0500
@@ -26,6 +26,7 @@
 
     private boolean func_154347_a(File p_154347_1_)
     {
+        if (true) return true; // Cauldron
         FileInputStream fileinputstream = null;
         boolean flag = false;
 
