--- eclipse/Clean/src/main/java/org/bukkit/command/defaults/BanIpCommand.java	2019-01-11 20:13:52.983930400 -0500
+++ eclipse/Cauldron/src/main/java/org/bukkit/command/defaults/BanIpCommand.java	2019-01-11 20:13:26.868531200 -0500
@@ -32,7 +32,7 @@
             return false;
         }
 
-        String reason = args.length > 0 ? StringUtils.join(args, ' ', 1, args.length) : null;
+        String reason = StringUtils.join(args, ' ', 1, args.length);
 
         if (ipValidity.matcher(args[0]).matches()) {
             processIPBan(args[0], sender, reason);
