--- eclipse/Clean/src/main/java/org/bukkit/command/defaults/PluginsCommand.java	2019-01-11 20:13:51.429906100 -0500
+++ eclipse/Cauldron/src/main/java/org/bukkit/command/defaults/PluginsCommand.java	2019-01-11 20:13:23.592587800 -0500
@@ -40,4 +40,12 @@
 
         return "(" + plugins.length + "): " + pluginList.toString();
     }
+
+    // Spigot Start
+    @Override
+    public java.util.List<String> tabComplete(CommandSender sender, String alias, String[] args) throws IllegalArgumentException
+    {
+        return java.util.Collections.emptyList();
+    }
+    // Spigot End
 }
\ No newline at end of file
