--- eclipse/Clean/src/main/java/org/bukkit/command/defaults/TellCommand.java	2019-01-11 20:13:53.262219600 -0500
+++ eclipse/Cauldron/src/main/java/org/bukkit/command/defaults/TellCommand.java	2019-01-11 20:13:27.411929900 -0500
@@ -45,4 +45,16 @@
 
         return true;
     }
+
+    // Spigot Start
+    @Override
+    public java.util.List<String> tabComplete(CommandSender sender, String alias, String[] args) throws IllegalArgumentException
+    {
+        if ( args.length == 0 )
+        {
+            return super.tabComplete( sender, alias, args );
+        }
+        return java.util.Collections.emptyList();
+    }
+    // Spigot End
 }
\ No newline at end of file
