--- eclipse/Clean/src/main/java/org/bukkit/command/defaults/WhitelistCommand.java	2019-01-11 20:13:51.277722300 -0500
+++ eclipse/Cauldron/src/main/java/org/bukkit/command/defaults/WhitelistCommand.java	2019-01-11 20:13:23.231975400 -0500
@@ -104,7 +104,7 @@
         } else if (args.length == 2) {
             if (args[0].equalsIgnoreCase("add")) {
                 List<String> completions = new ArrayList<String>();
-                for (OfflinePlayer player : Bukkit.getOfflinePlayers()) {
+                for (OfflinePlayer player : Bukkit.getOnlinePlayers()) {
                     String name = player.getName();
                     if (StringUtil.startsWithIgnoreCase(name, args[1]) && !player.isWhitelisted()) {
                         completions.add(name);
