--- eclipse/Clean/src/main/java/net/minecraft/world/gen/structure/MapGenStructure.java	2019-01-11 20:13:52.915853200 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/world/gen/structure/MapGenStructure.java	2019-01-11 20:13:26.817486500 -0500
@@ -239,8 +239,17 @@
     {
         if (this.field_143029_e == null)
         {
-            this.field_143029_e = (MapGenStructureData)p_143027_1_.perWorldStorage.loadData(MapGenStructureData.class, this.func_143025_a());
+            // Spigot Start
+            if (p_143027_1_.getSpigotConfig().saveStructureInfo && !this.func_143025_a().equals("Mineshaft")) // Cauldron
+            {
+                this.field_143029_e = (MapGenStructureData) p_143027_1_.loadItemData(MapGenStructureData.class, this.func_143025_a());
+            }
+            else
+            {
+                this.field_143029_e = new MapGenStructureData(this.func_143025_a());
+            }
 
+            // Spigot End
             if (this.field_143029_e == null)
             {
                 this.field_143029_e = new MapGenStructureData(this.func_143025_a());
