--- eclipse/Clean/src/main/java/net/minecraft/entity/projectile/EntitySnowball.java	2019-01-11 20:13:52.963916200 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/entity/projectile/EntitySnowball.java	2019-01-11 20:13:26.848509900 -0500
@@ -49,4 +49,11 @@
             this.setDead();
         }
     }
+
+    // Cauldron start
+    @Override
+    public boolean entityProjectileHook() {
+        return true;
+    }
+    // Cauldron end
 }
\ No newline at end of file
