--- eclipse/Clean/src/main/java/cpw/mods/fml/common/network/handshake/FMLHandshakeServerState.java	2019-01-11 20:13:52.692449200 -0500
+++ eclipse/Cauldron/src/main/java/cpw/mods/fml/common/network/handshake/FMLHandshakeServerState.java	2019-01-11 20:13:26.736429800 -0500
@@ -19,6 +19,10 @@
         {
             NetworkDispatcher dispatcher = ctx.channel().attr(NetworkDispatcher.FML_DISPATCHER).get();
             int overrideDim = dispatcher.serverInitiateHandshake();
+            if(overrideDim == -100000)
+            {
+            	return ERROR;
+            }
             ctx.writeAndFlush(FMLHandshakeMessage.makeCustomChannelRegistration(NetworkRegistry.INSTANCE.channelNamesFor(Side.SERVER))).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
             ctx.writeAndFlush(new FMLHandshakeMessage.ServerHello(overrideDim)).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
             return HELLO;
@@ -37,7 +41,11 @@
             }
 
             FMLHandshakeMessage.ModList client = (FMLHandshakeMessage.ModList)msg;
-            FMLLog.info("Client attempting to join with %d mods : %s", client.modListSize(), client.modListAsString());
+            if (net.minecraft.server.MinecraftServer.thermosConfig.loggingClientModList.getValue()) {
+                FMLLog.info("Client attempting to join with %d mods : %s", client.modListSize(), client.modListAsString());
+            } else {
+                FMLLog.info("Client attempting to join with %d mods", client.modListSize());
+            }
             String result = FMLNetworkHandler.checkModList(client, Side.CLIENT);
             if (result != null)
             {
