--- eclipse/Clean/src/main/java/net/minecraftforge/common/chunkio/ChunkIOExecutor.java	2019-01-11 20:13:51.796169500 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraftforge/common/chunkio/ChunkIOExecutor.java	2019-01-11 20:13:26.280108000 -0500
@@ -22,7 +22,7 @@
     }
 
     public static void adjustPoolSize(int players) {
-        int size = Math.max(BASE_THREADS, (int) Math.ceil(players / PLAYERS_PER_THREAD));
+        int size = Math.max(BASE_THREADS, (int) Math.ceil((double) players / PLAYERS_PER_THREAD));
         instance.setActiveThreads(size);
     }
 
