--- eclipse/Clean/src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java	2019-01-11 20:13:52.595092400 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java	2019-01-11 20:13:26.676387100 -0500
@@ -47,7 +47,7 @@
     public final int z;
     public final int face;
     public final World world;
-    
+    public org.bukkit.event.player.PlayerInteractEvent cb;
     public Result useBlock = DEFAULT;
     public Result useItem = DEFAULT;
 
