--- eclipse/Clean/src/main/java/net/minecraft/tileentity/TileEntityPiston.java	2019-01-11 20:13:52.242432400 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/tileentity/TileEntityPiston.java	2019-01-11 20:13:26.425209700 -0500
@@ -140,6 +140,11 @@
 
     public void updateEntity()
     {
+        if (this.worldObj == null)
+        {
+            return;    // CraftBukkit
+        }
+
         this.lastProgress = this.progress;
 
         if (this.lastProgress >= 1.0F)
