--- eclipse/Clean/src/main/java/net/minecraft/tileentity/TileEntityDropper.java	2019-01-11 20:13:51.952756500 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/tileentity/TileEntityDropper.java	2019-01-11 20:13:26.340150100 -0500
@@ -8,4 +8,12 @@
     {
         return this.hasCustomInventoryName() ? this.field_146020_a : "container.dropper";
     }
+
+    // Cauldron start
+    @Override
+    public boolean canUpdate()
+    {
+        return false;
+    }
+    // Cauldron end
 }
\ No newline at end of file
