--- eclipse/Clean/src/main/java/net/minecraft/server/management/UserListEntry.java	2019-01-11 20:13:52.383624200 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/server/management/UserListEntry.java	2019-01-11 20:13:26.516275100 -0500
@@ -17,7 +17,7 @@
         this.field_152642_a = p_i1147_1_;
     }
 
-    Object func_152640_f()
+    public Object func_152640_f() // CraftBukkit - private -> public
     {
         return this.field_152642_a;
     }
