--- eclipse/Clean/src/main/java/net/minecraft/server/management/UserList.java	2019-01-11 20:13:53.277230200 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/server/management/UserList.java	2019-01-11 20:13:27.440951000 -0500
@@ -184,6 +184,12 @@
         return this.field_152696_d.size() < 1;
     }
 
+    // CraftBukkit start
+    public Collection<UserListEntry> getValues() {
+        return this.field_152696_d.values();
+    }
+    // CraftBukkit end
+
     @SideOnly(Side.SERVER)
     public void func_152679_g() throws IOException
     {
