--- eclipse/Clean/src/main/java/net/minecraft/inventory/Slot.java	2019-01-11 20:13:51.269715200 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/inventory/Slot.java	2019-01-11 20:13:23.206802600 -0500
@@ -10,7 +10,7 @@
 
 public class Slot
 {
-    private final int slotIndex;
+    public final int slotIndex; // CraftBukkit - private -> public
     public final IInventory inventory;
     public int slotNumber;
     public int xDisplayPosition;
