--- eclipse/Clean/src/main/java/net/minecraft/village/Village.java	2019-01-11 20:13:50.623511000 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/village/Village.java	2019-01-11 20:13:22.456958000 -0500
@@ -71,7 +71,7 @@
             {
                 EntityIronGolem entityirongolem = new EntityIronGolem(this.worldObj);
                 entityirongolem.setPosition(vec3.xCoord, vec3.yCoord, vec3.zCoord);
-                this.worldObj.spawnEntityInWorld(entityirongolem);
+                this.worldObj.addEntity(entityirongolem, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.VILLAGE_DEFENSE); // CraftBukkit
                 ++this.numIronGolems;
             }
         }
