--- eclipse/Clean/src/main/java/net/minecraft/network/rcon/RConConsoleSource.java	2019-01-11 20:13:52.833436600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/network/rcon/RConConsoleSource.java	2019-01-11 20:13:26.775457100 -0500
@@ -25,6 +25,13 @@
         return new ChatComponentText(this.getCommandSenderName());
     }
 
+    // CraftBukkit start - Send a String
+    public void sendMessage(String message)
+    {
+        this.buffer.append(message);
+    }
+    // CraftBukkit end
+
     public void addChatMessage(IChatComponent p_145747_1_)
     {
         this.buffer.append(p_145747_1_.getUnformattedText());
