--- eclipse/Clean/src/main/java/net/minecraft/entity/ai/EntityAISit.java	2019-01-11 20:13:53.103780000 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/entity/ai/EntityAISit.java	2019-01-11 20:13:26.924636300 -0500
@@ -19,7 +19,7 @@
     {
         if (!this.theEntity.isTamed())
         {
-            return false;
+            return this.isSitting && this.theEntity.getAttackTarget() == null; // CraftBukkit - Allow sitting for wild animals
         }
         else if (this.theEntity.isInWater())
         {
