--- eclipse/Clean/src/main/java/net/minecraft/entity/ai/EntityAIPanic.java	2019-01-11 20:13:51.592133600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/entity/ai/EntityAIPanic.java	2019-01-11 20:13:23.917163500 -0500
@@ -3,6 +3,8 @@
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.util.Vec3;
 
+import net.minecraft.entity.EntityLivingBase; // CraftBukkit
+
 public class EntityAIPanic extends EntityAIBase
 {
     private EntityCreature theEntityCreature;
@@ -50,6 +52,14 @@
 
     public boolean continueExecuting()
     {
+        // CraftBukkit start - introduce a temporary timeout hack until this is fixed properly
+        if ((this.theEntityCreature.ticksExisted - this.theEntityCreature.func_142015_aE()) > 100)
+        {
+            this.theEntityCreature.setRevengeTarget((EntityLivingBase) null);
+            return false;
+        }
+
+        // CraftBukkit end
         return !this.theEntityCreature.getNavigator().noPath();
     }
 }
\ No newline at end of file
