--- eclipse/Clean/src/main/java/net/minecraft/entity/ai/EntityAITasks.java	2019-01-11 20:13:52.294519300 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/entity/ai/EntityAITasks.java	2019-01-11 20:13:26.460234400 -0500
@@ -104,7 +104,10 @@
         while (iterator.hasNext())
         {
             entityaitaskentry = (EntityAITasks.EntityAITaskEntry)iterator.next();
-            this.theProfiler.startSection(entityaitaskentry.action.getClass().getSimpleName());
+            if (this.theProfiler.profilingEnabled) //Unnecessary lag...do it only if profiling is on
+            	this.theProfiler.startSection(entityaitaskentry.action.getClass().getSimpleName());
+            else
+            	this.theProfiler.startSection("EntityAITaskEntry.EntityAIBase");
             entityaitaskentry.action.startExecuting();
             this.theProfiler.endSection();
         }
