--- eclipse/Clean/src/main/java/net/minecraft/entity/item/EntityExpBottle.java	2019-01-11 20:13:52.250440300 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/entity/item/EntityExpBottle.java	2019-01-11 20:13:26.435217600 -0500
@@ -43,8 +43,19 @@
     {
         if (!this.worldObj.isRemote)
         {
-            this.worldObj.playAuxSFX(2002, (int)Math.round(this.posX), (int)Math.round(this.posY), (int)Math.round(this.posZ), 0);
+            // CraftBukkit moved after event
+            //this.worldObj.playAuxSFX(2002, (int)Math.round(this.posX), (int)Math.round(this.posY), (int)Math.round(this.posZ), 0);
             int i = 3 + this.worldObj.rand.nextInt(5) + this.worldObj.rand.nextInt(5);
+            // CraftBukkit start
+            org.bukkit.event.entity.ExpBottleEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callExpBottleEvent(this, i);
+            i = event.getExperience();
+
+            if (event.getShowEffect())
+            {
+                this.worldObj.playAuxSFX(2002, (int) Math.round(this.posX), (int) Math.round(this.posY), (int) Math.round(this.posZ), 0);
+            }
+
+            // CraftBukkit end
 
             while (i > 0)
             {
@@ -56,4 +67,11 @@
             this.setDead();
         }
     }
+
+    // Cauldron start
+    @Override
+    public boolean entityProjectileHook() {
+        return true;
+    }
+    // Cauldron end
 }
\ No newline at end of file
