--- eclipse/Clean/src/main/java/net/minecraft/world/gen/FlatLayerInfo.java	2019-01-11 20:13:53.184102600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/world/gen/FlatLayerInfo.java	2019-01-11 20:13:26.959661400 -0500
@@ -12,7 +12,6 @@
 
     public FlatLayerInfo(int p_i45467_1_, Block p_i45467_2_)
     {
-        this.layerCount = 1;
         this.layerCount = p_i45467_1_;
         this.field_151537_a = p_i45467_2_;
     }
