--- eclipse/Clean/src/main/java/net/minecraft/world/WorldManager.java	2019-01-11 20:13:52.318536600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/world/WorldManager.java	2019-01-11 20:13:26.473244200 -0500
@@ -12,7 +12,7 @@
 public class WorldManager implements IWorldAccess
 {
     private MinecraftServer mcServer;
-    private WorldServer theWorldServer;
+    public WorldServer theWorldServer; // CraftBukkit - private -> public
     private static final String __OBFID = "CL_00001433";
 
     public WorldManager(MinecraftServer p_i1517_1_, WorldServer p_i1517_2_)
