--- eclipse/Clean/src/main/java/net/minecraft/block/BlockLiquid.java	2019-01-11 20:13:52.629239000 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/block/BlockLiquid.java	2019-01-11 20:13:26.694400700 -0500
@@ -509,7 +509,7 @@
                     {
                         p_149805_1_.setBlock(p_149805_2_, p_149805_3_, p_149805_4_, Blocks.obsidian);
                     }
-                    else if (l <= 4)
+                    else if (l > 0) //Paper Spigot
                     {
                         p_149805_1_.setBlock(p_149805_2_, p_149805_3_, p_149805_4_, Blocks.cobblestone);
                     }
