--- eclipse/Clean/src/main/java/net/minecraft/block/BlockSnow.java	2019-01-11 20:13:51.936745400 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/block/BlockSnow.java	2019-01-11 20:13:26.333144900 -0500
@@ -114,6 +114,14 @@
     {
         if (p_149674_1_.getSavedLightValue(EnumSkyBlock.Block, p_149674_2_, p_149674_3_, p_149674_4_) > 11)
         {
+            // CraftBukkit start
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockFadeEvent(p_149674_1_.getWorld().getBlockAt(p_149674_2_, p_149674_3_, p_149674_4_), Blocks.air).isCancelled())
+            {
+                return;
+            }
+
+            // CraftBukkit end
+            this.dropBlockAsItem(p_149674_1_, p_149674_2_, p_149674_3_, p_149674_4_, p_149674_1_.getBlockMetadata(p_149674_2_, p_149674_3_, p_149674_4_), 0);
             p_149674_1_.setBlockToAir(p_149674_2_, p_149674_3_, p_149674_4_);
         }
     }
