--- eclipse/Clean/src/main/java/net/minecraft/block/BlockDaylightDetector.java	2019-01-11 20:13:52.567072600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/block/BlockDaylightDetector.java	2019-01-11 20:13:26.657378300 -0500
@@ -73,6 +73,7 @@
 
             if (l != i1)
             {
+                i1 = org.bukkit.craftbukkit.event.CraftEventFactory.callRedstoneChange(p_149957_1_, p_149957_2_, p_149957_3_, p_149957_4_, l, i1).getNewCurrent(); // CraftBukkit - Call BlockRedstoneEvent
                 p_149957_1_.setBlockMetadataWithNotify(p_149957_2_, p_149957_3_, p_149957_4_, i1, 3);
             }
         }
