--- eclipse/Clean/src/main/java/net/minecraft/item/crafting/IRecipe.java	2019-01-11 20:13:52.636268500 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/item/crafting/IRecipe.java	2019-01-11 20:13:26.699404000 -0500
@@ -13,4 +13,6 @@
     int getRecipeSize();
 
     ItemStack getRecipeOutput();
+
+    org.bukkit.inventory.Recipe toBukkitRecipe(); // CraftBukkit
 }
\ No newline at end of file
