--- eclipse/Clean/src/main/java/net/minecraft/network/Packet.java	2019-01-11 20:13:53.154861400 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/network/Packet.java	2019-01-11 20:13:26.946652200 -0500
@@ -9,6 +9,7 @@
 public abstract class Packet
 {
     private static final Logger logger = LogManager.getLogger();
+    public final long timestamp = System.currentTimeMillis(); // CraftBukkit
     private static final String __OBFID = "CL_00001272";
 
     public static Packet generatePacket(BiMap p_148839_0_, int p_148839_1_)
