--- eclipse/Clean/src/main/java/net/minecraft/network/PacketBuffer.java	2019-01-11 20:13:50.644525600 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/network/PacketBuffer.java	2019-01-11 20:13:22.460966300 -0500
@@ -98,7 +98,7 @@
 
     public void writeItemStackToBuffer(ItemStack p_150788_1_) throws IOException
     {
-        if (p_150788_1_ == null)
+        if (p_150788_1_ == null || p_150788_1_.getItem() == null)   // CraftBukkit - NPE fix itemstack.getItem()
         {
             this.writeShort(-1);
         }
