--- eclipse/Clean/src/main/java/org/bukkit/entity/Entity.java	2019-01-11 20:13:53.323315400 -0500
+++ eclipse/Cauldron/src/main/java/org/bukkit/entity/Entity.java	2019-01-11 20:13:27.483980200 -0500
@@ -291,4 +291,22 @@
      * @return The current vehicle.
      */
     public Entity getVehicle();
+
+    // Spigot Start
+    public class Spigot
+    {
+
+        /**
+         * Returns whether this entity is invulnerable.
+         *         
+        * @return True if the entity is invulnerable.
+         */
+        public boolean isInvulnerable()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+    }
+
+    Spigot spigot();
+    // Spigot End
 }
\ No newline at end of file
