--- eclipse/Clean/src/main/java/net/minecraft/network/play/server/S05PacketSpawnPosition.java	2019-01-11 20:13:50.461903700 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraft/network/play/server/S05PacketSpawnPosition.java	2019-01-11 20:13:22.421933300 -0500
@@ -10,9 +10,9 @@
 
 public class S05PacketSpawnPosition extends Packet
 {
-    private int field_149364_a;
-    private int field_149362_b;
-    private int field_149363_c;
+    public int field_149364_a; // CraftBukkit - private -> public
+    public int field_149362_b; // CraftBukkit - private -> public
+    public int field_149363_c; // CraftBukkit - private -> public
     private static final String __OBFID = "CL_00001336";
 
     public S05PacketSpawnPosition() {}
