--- eclipse/Clean/src/main/java/net/minecraftforge/common/ForgeChunkManager.java	2019-01-11 20:13:52.859717500 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraftforge/common/ForgeChunkManager.java	2019-01-11 20:13:26.782462400 -0500
@@ -352,6 +352,7 @@
         {
             this.ticket = ticket;
             this.location = location;
+            ticket.world.activity.put(location, ticket);
         }
     }
 
@@ -363,6 +364,7 @@
         {
             this.ticket = ticket;
             this.location = location;
+            ticket.world.activity.remove(location, ticket);
         }
     }
 
