--- eclipse/Clean/src/main/java/net/minecraftforge/common/ForgeVersion.java	2019-01-11 20:13:52.326545900 -0500
+++ eclipse/Cauldron/src/main/java/net/minecraftforge/common/ForgeVersion.java	2019-01-11 20:13:26.475246400 -0500
@@ -25,7 +25,7 @@
     //This number is incremented every time a interface changes or new major feature is added, and reset every Minecraft version
     public static final int revisionVersion = 4;
     //This number is incremented every time Jenkins builds Forge, and never reset. Should always be 0 in the repo code.
-    public static final int buildVersion    = 0;
+    public static final int buildVersion    = thermos.Thermos.lookupForgeRevision(); // KCauldron
 
     private static Status status = PENDING;
     private static String target = null;
@@ -145,4 +145,4 @@
             }
         }.start();
     }
-}
+}
\ No newline at end of file
